cirq/sim/simulator.py

```diff
@@ -372,6 +374,33 @@ def simulate_sweep(
         """
         raise NotImplementedError()
 
+    def simulate_expectation_values_sweep(
+        self,
+        program: 'cirq.Circuit',
+        observables: Union['cirq.PauliSumLike', List['cirq.PauliSumLike']],
+        params: 'study.Sweepable',
+        qubit_order: ops.QubitOrderOrList = ops.QubitOrder.DEFAULT,
+        initial_state: Any = None,
+        permit_terminal_measurements: bool = False,
+    ) -> List[List[float]]:
+        if not permit_terminal_measurements and program.are_any_measurements_terminal():
+            raise ValueError(
+                'Provided circuit has terminal measurements, which may '
+                'skew expectation values. If this is intentional, set '
+                'permit_terminal_measurements=True.'
+            )
+        swept_evs = []
+        qubit_order = ops.QubitOrder.as_qubit_order(qubit_order)
+        if not isinstance(observables, List):
+            observables = [observables]
+        pslist = [ops.PauliSum.wrap(pslike) for pslike in observables]
+        for param_resolver in study.to_resolvers(params):
+            result = self.simulate(
+                program, param_resolver, qubit_order=qubit_order, initial_state=initial_state
+            )
+            swept_evs.append([result.expectation_from_state(obs) for obs in pslist])
```

Comment:
My concern here: the fact that `expectation_from_state` has no default implementation means that `SimulatesFullState` does not fully implement `SimulatesExpectationValues`, which means user-created subclasses will have a broken API.

We could update the class docstring to say that users must implement `expectation_from_state` in their result class, but that's kind of unwieldy for a class whose base purpose does not include calculating expectation values. This is part of why I mentioned needing a `SimulatesEVsAndFullState` type for this in #3979.